### PR TITLE
Add ability to use async library loader

### PR DESCRIPTION
**Pull request by danielrudn**
Reopening #61 after working on the async library loader.

The solution I used was to load all libraries into memory (order doesn't matter at this point) and then synchronously load the dependencies in the correct order using the libraries in memory.

Related is #47 and #53

### CODE REVIEW

**Comment by JPSchellenberg** [comment]
Hey, 
thank you for your PR!
I fixed the code-style in the content-test. 
Unfortunately, the dependency-order is not correct yet. 
I added a content-test for the async-libraryLoader (download all content-types with `npm run download:content` and then run the test with `npm run test:content:async`) and as you can see it fails due to a incorrect depedency-order. I did not inspect your code yet, but I hope I will get to it the next days.

Screenshot 2019-08-06 at 21 00 51

I will keep this PR open and hope we will get the async-libraryLoader working.. :)

**Comment by JPSchellenberg** [comment]
It also seems to break the sync-libraryLoader: https://travis-ci.com/Lumieducation/H5P-Nodejs-library/builds/122169503#L3041

**Comment by danielrudn** [comment]
Hey thanks for the quick reply!

I looked into it and it seems that the [Impressive Presentation content-type](https://h5p.org/impressive-presentation) is causing issues. It comes with `FontAwesome-4.3` but one of the dependencies (`H5P.SingleChoiceSet-1.3`) tries to preload `FontAwesome-4.5` which doesn't exist. If I remove the Impress Presentation content-type from the `examples.json` all of the tests pass for both sync and async. I can have the tests ignore that specific content-type for now but i'm not sure if that's the right thing to do.

**Comment by JPSchellenberg** [comment]
You are right.. as the impressive presentation seems to be in an alpha-phase I would suggest to omit it. I will exclude it from the examples tomorrow and then the PR should be ready to merge.
Thank you very much for your contribution, as your contribution seems to solve a crucial [bug in the Editor](https://github.com/Lumieducation/H5P-Editor-Nodejs-library/issues/47)  as well.

**Comment by rtens** [comment]
Please let me know when the conflict is resolved.

File: examples/scrape-examples.js [hunk 1]
```diff
@@ -1,13 +1,13 @@
 const request = require('request-promise');
-const { JSDOM } = require("jsdom");
+const { JSDOM } = require('jsdom');
```
[the remark this answers is not in the document]
Comment (JPSchellenberg) [reply examples/scrape-examples.js]: It was the plugin in my IDE. I will turn it off and only run `npm run format`.

**Pull request merged**